src/na/device/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+# Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+# Copyright (c) 2025 Munich Quantum Software Company GmbH
+# All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+#
+# Licensed under the MIT License
+
+# Set target name
+set(TARGET_NAME ${MQT_CORE_TARGET_NAME}-na-device-gen)
+
+# If the target is not already defined
+if(NOT TARGET ${TARGET_NAME})
+
+  # Add library for device generation
+  add_mqt_core_library(${TARGET_NAME})
+
+  # add sources to target
+  target_sources(${TARGET_NAME} PRIVATE Generator.cpp)
+
+  # add headers using file sets
+  target_sources(${TARGET_NAME} PUBLIC FILE_SET HEADERS BASE_DIRS ${MQT_CORE_INCLUDE_BUILD_DIR}
+                                       FILES ${MQT_CORE_INCLUDE_BUILD_DIR}/na/device/Generator.hpp)
+
+  # Link nlohmann_json, spdlog
+  target_link_libraries(
+    ${TARGET_NAME}
+    PUBLIC nlohmann_json::nlohmann_json
+    PRIVATE spdlog::spdlog)
+
+  # set versioning information
+  set_target_properties(
+    ${TARGET_NAME}
+    PROPERTIES VERSION ${PROJECT_VERSION}
+               SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
+               EXPORT_NAME CoreNaDeviceGen)
+
+  # set c++ standard
+  target_compile_features(${TARGET_NAME} PRIVATE cxx_std_17)
+
+  # Make version available
+  target_compile_definitions(${TARGET_NAME} PRIVATE MQT_CORE_VERSION="${MQT_CORE_VERSION}")
+endif()
+
+# Set target name
+set(TARGET_NAME mqt-core-qdmi-na-device-generator)
+
+# If the target is not already defined
+if(NOT TARGET ${TARGET_NAME})
+  # Add executable for device generation
+  add_executable(${TARGET_NAME})
+
+  # add sources to target
+  target_sources(${TARGET_NAME} PRIVATE App.cpp)
+
+  # Link Generator library
+  target_link_libraries(${TARGET_NAME} PRIVATE MQT::CoreNaDeviceGen spdlog::spdlog)
+
+  # set versioning information
+  set_target_properties(${TARGET_NAME} PROPERTIES VERSION ${PROJECT_VERSION} EXPORT_NAME
+                                                                             CoreNaDeviceGen)
+  # place in the bin directory
+  set_target_properties(${TARGET_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY
+                                                  "${CMAKE_BINARY_DIR}/bin")
+
+  # set c++ standard
+  target_compile_features(${TARGET_NAME} PRIVATE cxx_std_17)
+
+  # Make version available
+  target_compile_definitions(${TARGET_NAME} PRIVATE MQT_CORE_VERSION="${MQT_CORE_VERSION}")
+
+  # Create an alias for the target
+  add_executable(MQT::CoreQdmiNaDeviceGenerator ALIAS ${TARGET_NAME})
```

Comment:
Should. We always capitalize QDMI?
Also, none of the other target names or aliases above contain QDMI. Maybe all of them should.